### PR TITLE
Robustness fixes

### PR DESCRIPTION
**Pull request by jeffamstutz**
I added a few commits that avoids dereferencing null objects where applicable, and creates an index array for triangle geometry when one isn't provided by the application (this is valid).

### CODE REVIEW

**Comment by jeffamstutz** [comment]
I also experimented a bit with `Frame::renderFrame()` -- before the device would only render one of the surface objects, so collecting them into a single `vtkm::rendering::Scene` seemed to fix that.

**Comment by jeffamstutz** [comment]
Here's a screenshot of it running beside VisRTX in one of my test applications, where the scene consists of 5 separate surface objects each with their own unique mesh:

image

File: device/frame/Frame.cpp [hunk 1]
```diff
@@ -221,47 +221,55 @@ void Frame::renderFrame()
...
+                ANARI_SEVERITY_DEBUG, "skip rendering invalid volume");
+            continue;
+          }
+          volume_scene.AddActor(*volume->actor());
         }
       }
+
+      if (surface_scene.GetNumberOfActors() != 0) {
+        vtkm::rendering::View3D surface_view(surface_scene,
+            vtkm::rendering::MapperRayTracer(),
```
Comment (kmorel) [inline device/frame/Frame.cpp:254]: This is fine for now, but this is indicative of us having to shift things around later. At some point we should support other surface geometry like spheres (which are supported by the VTK-m raycaster). The problem is that the structure of these outward facing classes do a bad job of mixing mapper types, which is what you really want here. I think there is a way around that, but we may eschew them altogether and interface directly with the raycast classes.

File: device/scene/surface/geometry/Geometry.cpp [hunk 2]
```diff
@@ -7,21 +7,13 @@
 
 #include "array/ArrayConversion.h"
 
-namespace {
-
-} // anonymous namespace
+namespace {} // anonymous namespace
```
Comment (kmorel) [inline device/scene/surface/geometry/Geometry.cpp]: We should just delete this empty namespace.

**Review by kmorel — APPROVED** [review]
Thanks for taking a pass and cleaning up our code.

**Comment by dpugmire** [comment]
Thanks @jeffamstutz !

**Pull request merged**